test/javascripts/acceptance/timezone-offset-test.js

```diff
@@ -0,0 +1,358 @@
+    { id: 7, count: 0, hidden: false, can_act: true },
+  ],
+  chunk_size: 20,
+  bookmarked: false,
+  bookmarks: [],
+  topic_timer: null,
+  message_bus_last_id: 16,
+  participant_count: 1,
+  show_read_indicator: false,
+  thumbnails: null,
+  slow_mode_enabled_until: null,
+  summarizable: false,
+  details: {
+    can_edit: true,
+    notification_level: 3,
+    notifications_reason_id: 1,
+    can_move_posts: true,
+    can_delete: true,
+    can_remove_allowed_users: true,
+    can_invite_to: true,
+    can_invite_via_email: true,
+    can_create_post: true,
+    can_reply_as_new_topic: true,
+    can_flag_topic: true,
+    can_convert_topic: true,
+    can_review_topic: true,
+    can_close_topic: true,
+    can_archive_topic: true,
+    can_split_merge_topic: true,
+    can_edit_staff_notes: true,
+    can_toggle_topic_visibility: true,
+    can_pin_unpin_topic: true,
+    can_moderate_category: true,
+    can_remove_self_id: 1,
+    participants: [
+      {
+        id: 1,
+        username: "jan",
+        name: null,
+        avatar_template: "/letter_avatar_proxy/v4/letter/j/ce7236/{size}.png",
+        post_count: 4,
+        primary_group_name: null,
+        flair_name: null,
+        flair_url: null,
+        flair_color: null,
+        flair_bg_color: null,
+        flair_group_id: null,
+        admin: true,
+        trust_level: 1,
+      },
+    ],
+    created_by: {
+      id: 1,
+      username: "jan",
+      name: null,
+      avatar_template: "/letter_avatar_proxy/v4/letter/j/ce7236/{size}.png",
+    },
+    last_poster: {
+      id: 1,
+      username: "jan",
+      name: null,
+      avatar_template: "/letter_avatar_proxy/v4/letter/j/ce7236/{size}.png",
+    },
+  },
+};
+
+function getEventByText(text) {
+  const events = [...document.querySelectorAll(".fc-day-grid-event")].filter(
+    (event) => event.textContent.includes(text)
+  );
+  return events.length === 1 ? events[0] : events;
+}
+
+acceptance("Discourse Calendar - Timezone Offset", function (needs) {
+  needs.settings({
+    calendar_enabled: true,
+    enable_timezone_offset_for_calendar_events: true,
+  });
+
+  needs.pretender((server, helper) => {
+    server.get("/t/252.json", () => {
+      return helper.response(topicResponse);
+    });
+  });
+
+  test("doesn't apply an offset for events in the same timezone", async (assert) => {
+    await visit("/t/252");
+
+    const eventElement = getEventByText("Lisbon");
+
+    assert.notOk(eventElement.style.marginLeft);
+    assert.notOk(eventElement.style.marginRight);
+  });
+
+  test("applies the correct offset for events that extend into the next day", async (assert) => {
+    await visit("/t/252");
+
+    const eventElement = getEventByText("Cordoba");
+
+    assert.ok(eventElement.style.marginLeft.includes("8.33")); // ( ( 1 - (-3) ) / 24 ) * 50%
+    assert.ok(eventElement.style.marginRight.includes("41.66")); // ( ( 24 - ( 1 - (-3) ) ) / 24 ) * 50%
```

Comment:
This is cool but I am a little worried it's going to be fragile. Different browsers may be off by a hair, and the test will fail. CSS changes will also likely make it fail (the 1 and -3 values come from the stylesheet, right?). 

I'd be ok with making this less precise. Either by checking that the margin matches the rounded value (i.e. == 8) or simply checking the presence of a margin? It's certainly not as accurate, but the test will be a bit more resilient.

Reply:
Yes, I agree, it can fail at some point, but checking only for the margin presense feels like we are only testing one conditional. Checking the round value is better. We need to parse the string to get the value, but it's not that bad.
I'll update the test before merging.

1 and -3 correspond to the timezones, Lisbon (GMT+1), Cordoba (GMT-3)